src/ripple/protocol/PublicKey.h

```diff
@@ -61,13 +61,41 @@ namespace ripple {
 class PublicKey
 {
 protected:
-    std::size_t size_ = 0;
-    std::uint8_t buf_[33];  // should be large enough
+    // All the constructed public keys are valid, non-empty and contain 33
+    // bytes of data.
+    static constexpr std::size_t size_ = 33;
```

Comment:
Now that `size_` is a `constexpr`, any checks of it don't make sense. I've already pointed out issues in some of the ctors, but I noticed some more.
* It's impossible for `empty()` to return `true`. It should be removed entirely.
* Any remaining uses of `size_` should be checked to ensure they make sense. I don't think there are many.